### ISSUE TITLE
Autofill not happening on some sites

### ISSUE THREAD

**Issue by belamu** [passff/passff#507]
## Environment
Firefox 98.0 on Fedora KDE

### Versions
- Operating system: Fedora KDE
- Browser: Firefox 98.0
- PassFF: Version 1.12
- Host app: pass 1.7.4.

Status line output:
`[09:00:41] show -> (0) no error message`

### Observations
For some sites login credentials are not autofilled, while for most they are. Problematic sites include
- https://www.trustroots.org/signin
  - login field id: username name: username
  - password field id: password name: password
- https://www.gls-online-filiale.de/services_cloud/portal
  - login field id: vrNetKey name: vrNetKeyFormControl
  - password field id: pin name: pinFormControl
- https://account.komoot.com/signin
  - login field id: email name: email
When I trigger the filling manually, e.g. by Ctrl+y, Enter or the menu or the right-click menu, the fields are filled as expected.
The passff icon (P on grey circle) is displayed in login field and password field.

I could not find a difference between those sites and others.

#### Preferences
*Autofilling: Try to auto fill page login forms* - yes
*Try to autosubmit page login forms* - no
*Password field names:* passwd,password,pass,questionAnswer,view:pin:pin
*Login field names:* login,user,mail,email,username,opt_login,log,usr_name,identifier,txtBenutzerkennung,idb60a227,view:zugangskennung:zugangskennung,u,cardNumber,text,bor_id,lbis_text_box_line,vrNetKey

#### Web console
(parts that seemed to have connection to passff):
```
09:11:23.819 [PassFF.content] 1 matches for https://www.gls-online-filiale.de/services_auth/auth-ui/?v=d…-filiale.de%2Fservices_cloud%2Fportal%2Fportal-oauth%2Flogin util.js:84:14
09:11:23.820 [PassFF.content] Best fit item geld/gls-online-filiale.de for https://www.gls-online-filiale.de/services_auth/auth-ui/?v=d…-filiale.de%2Fservices_cloud%2Fportal%2Fportal-oauth%2Flogin util.js:84:14
09:11:24.059 [PassFF.content] Start pref-auto-fill util.js:84:14
09:11:24.062 [PassFF.content] fillInputs: No relevant login input elements recognized. util.js:84:14
09:11:24.070 TypeError: PassFF.Page.fillInputs(...) is null
...
09:11:24.710 [PassFF.content] Inject icon pin
```
for trustroots:
```
[PassFF.content] 1 matches for https://www.trustroots.org/signin util.js:84:14
[PassFF.content] Best fit item websites/trustroots.org for https://www.trustroots.org/signin util.js:84:14
[PassFF.content] Inject icon username util.js:84:14
[PassFF.content] Inject icon password util.js:84:14
```
---

### Steps to reproduce the issue
1. Open one of the mentioned urls
2. See empty login fields that stay empty. No passphrase prompt happens.
3. Ctrl+y, down, enter. (only one option) Fields are filled (after passphrase prompt if necessary)

### Actual behaviour
- fields that stay empty. No password prompt happens.

### Expected behaviour
- as with other websites fields are autofilled (and not submitted in my case)

### Urgency
this has low priority because the extra work for problematic sites is quite limited (3 keystrokes). But I find it quite confusing: what do those websites differently?

**Comment by tuxor1337** [comment]
Thanks for the detailed report!

To summarize: For the sites you listed, when manually instructing PassFF to fill the form, everything works fine. But the form is not automatically filled on load.

We had a similar issue before, and there the problem was that the form's HTML was only injected into the site after the document has finished loading: #480.

I said it in #480, and I will stick to this: I tag this issue with "won't fix" because I personally consider the auto-fill feature a security issue in PassFF and don't recommend using it. Hence, I don't want to work on a fix. But if you would like to work on a solution, feel free to open a pull request.

**Comment by belamu** [comment]
Thank you for the explanation. I guess I didn't find the other issue since it's closed. I don't know how to prevent that different people report the same thing again.
> I said it in #480, and I will stick to this: I tag this issue with "won't fix" because I personally consider the auto-fill feature a security issue in PassFF and don't recommend using it. Hence, I don't want to work on a fix. But if you would like to work on a solution, feel free to open a pull request.

Makes sense!

**Comment by tuxor1337** [comment]
No, it's not your fault. If you like the feature feel free to use it, and even enhance it. It's just me setting personal priorities here :) I will pin this issue for future reference now.
